modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamIndexSettingsProvider.java

```diff
@@ -161,19 +164,34 @@ private List<String> findRoutingPaths(String indexName, Settings allSettings, Li
 
             List<String> routingPaths = null;
             for (var fieldMapper : mapperService.documentMapper().mappers().fieldMappers()) {
-                if (fieldMapper instanceof KeywordFieldMapper keywordFieldMapper) {
-                    if (keywordFieldMapper.fieldType().isDimension()) {
-                        if (routingPaths == null) {
-                            routingPaths = new ArrayList<>();
-                        }
-                        routingPaths.add(keywordFieldMapper.name());
-                    }
-                }
+                routingPaths = extractPath(routingPaths, fieldMapper);
+            }
+            for (var template : mapperService.getAllDynamicTemplates()) {
+                var templateName = "__dynamic__" + template.name();
```

Comment:
Where does the `__dynamic__` come from? Is that a standard naming convention somewhere?

Reply:
Not really a convention, but I the same template name being used when validating templates in `RootObjectMapper#validate(...)` method.